modules/javafx.graphics/src/main/java/javafx/scene/Parent.java

```diff
@@ -292,6 +292,10 @@ private void computeViewOrderChidrenAndUpdatePeer() {
     // Call this method if children view order is needed for picking.
     // The returned list should be treated as read only.
     private List<Node> getOrderedChildren() {
+        if (isDirty(DirtyBits.PARENT_CHILDREN_VIEW_ORDER)) {
+            //Fix for JDK-8205092
+            computeViewOrderChildrenAndUpdatePeer();
```

Comment:
As mentioned in my earlier comment, this needs to be changed as it is not thread-safe.

Reply:
Thanks for the detailed explanation. I went with the second option, although I didn't make a second method to update the peer's view order since it was a one-liner.